src/include/btree_cmp_inline.h

```diff
@@ -41,51 +187,22 @@ __wt_lex_compare(const WT_ITEM *user_item, const WT_ITEM *tree_item)
     userp = (const uint8_t *)user_item->data;
     treep = (const uint8_t *)tree_item->data;
 
-#ifdef HAVE_X86INTRIN_H
     /* Use vector instructions if we'll execute at least 2 of them. */
     if (len >= WT_VECTOR_SIZE * 2) {
         size_t remain;
         __m128i res_eq, u, t;
 
-        remain = len % WT_VECTOR_SIZE;
-        len -= remain;
-        if (WT_ALIGNED_16(userp) && WT_ALIGNED_16(treep))
-            for (; len > 0;
-                 len -= WT_VECTOR_SIZE, userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
-                u = _mm_load_si128((const __m128i *)userp);
-                t = _mm_load_si128((const __m128i *)treep);
-                res_eq = _mm_cmpeq_epi8(u, t);
-                if (_mm_movemask_epi8(res_eq) != 65535)
-                    break;
-            }
-        else
-            for (; len > 0;
-                 len -= WT_VECTOR_SIZE, userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
-                u = _mm_loadu_si128((const __m128i *)userp);
-                t = _mm_loadu_si128((const __m128i *)treep);
-                res_eq = _mm_cmpeq_epi8(u, t);
-                if (_mm_movemask_epi8(res_eq) != 65535)
-                    break;
-            }
-        len += remain;
-    }
-#elif defined(HAVE_ARM_NEON_INTRIN_H)
```

Comment:
With this PR we no longer use `HAVE_ARM_NEON_INTRIN_H`. Please also remove it from CMake.

Is `ldp` a 1 for 1 replacement of the Neon intrinsics? I'm wondering if there are platforms that would benefit from Neon but not from `ldp`.

Reply:
done

Reply:
The CMake changes LGTM. I'll leave this comment chain open in case others have thoughts on platforms with Neon but not `ldp`

Reply:
We probably _should_ be using it rather than inline asm to compute the crc32c, but perhaps this isn't the right place to discuss that.

> platforms with Neon but not ldp

I think that is just legacy arm32 (although it did have a different way to load multiple registers), so not worth caring about.